tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -917,6 +933,15 @@ private void doCommit(long transactionId, long writePointer, Set<ChangeId> chang
         invalidArray = invalid.toLongArray();
         LOG.info("Tx invalid list: removed committed tx {}", transactionId);
       }
+    } else {
+      long[] checkpointPointers = previous.getCheckpointWritePointers().toLongArray();
+      if (checkpointPointers != null && checkpointPointers.length > 0) {
```

Comment:
I can't find a away that this could be null. Changed it everywhere it is used to assume it is not null